packages/jellyfish-api-core/src/category/wallet.ts

```diff
@@ -325,6 +325,21 @@ export class Wallet {
     return await this.client.call('listwallets', [], 'number')
   }
 
+  /**
+   * If a label name is provided, this will return only incoming transactions paying to addresses with the specified label.
+   * Returns up to 'count' most recent transactions skipping the first 'from' transactions.
+   *
+   * @param {string} label If set, should be a valid label name to return only incoming transactions with the specified label, or '*' to disable filtering and return all transactions. (default = '*')
+   * @param {string} count The number of transactions to return (default = 10)
+   * @param {string} skip The number of transactions to skip (default = 0)
+   * @param {boolean} includeWatchOnly Whether to include watch-only addresses (default = true)
+   * @param {boolean} excludeCustomTx False to include all transactions, otherwise exclude custom transactions (default = false)
+   * @return {Promise<Array<InWalletTransactionWithFlatDetails>>}
+   */
+  async listTransactions (label: string = '*', count: number = 10, skip: number = 0, includeWatchOnly: boolean = true, excludeCustomTx: boolean = false): Promise<InWalletTransactionWithFlatDetails[]> {
```

Comment:
Since there are so many default params, should we use an object (typescript's equivalent for named parameters)?

Reply:
sure, structure will be like this:

`{ label?: string, count?: number, skip?: number, includeWatchOnly?: boolean, excludeCustomTx?: boolean }`

Reply:
Seeing some additional returned optional parameters being returned from the RPC call that are not part of the specifications from the `.cpp` code

https://github.com/DeFiCh/ain/blob/master/src/wallet/rpcwallet.cpp
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/15076300/220548827-f5a430a3-b6c9-4de6-b88b-ab1e89ea8b22.png">

Additional optional returned parameters include:
- `generated` -> boolean
- `walletconflicts` -> array

These parameters would only appear in some results, being non-existent in others

Using `toStrictEqual` will cause the tests to fail when the `generated` & `walletconflicts` parameters appear

`toMatchObject` only checks that a JavaScript object matches a subset of the properties of an object